### ISSUE TITLE
N806 special case variables of type `type` or `type[SOMEBASE]`

### ISSUE THREAD

**Issue by y2kbugger** [astral-sh/ruff#24817]
### Summary

```python
# ruff N806 false-positive: loop variable holding a type

class Base:
    pass

class Foo(Base):
    pass

class Bar(Base):
    pass

models: list[type[Base]] = [Foo, Bar]

# N806: Variable `Model` in function should be lowercase
# But `Model` is a type, not an instance — uppercase is the universal convention.
def process() -> None:
    for Model in models:  # (N806 here, side note, url fragment and copy link in playground is not working from brave browser for me right now, even in private more)
        print(Model.__name__)
```

**Comment by y2kbugger** [comment]
See https://github.com/astral-sh/ruff/issues/1985 for similar precedent

**Comment by ntBre** [comment]
I think this is quite a bit different from #1985, where we can directly see that the assignment was a `TypeVar` (or `NewType`). I don't think there's a robust way we could identify that `models` yields types without having access to type inference from ty. This seems like a good case for `noqa` or just to use the name `model`, which is what I would probably do personally.
